app/logic/fileHandler.py

```diff
@@ -43,25 +49,33 @@ def saveFiles(self, saveOriginalFile=None):
         """ Saves the attachment in the app/static/files/eventattachments/ or courseattachements/ directory """
         try:
             for file in self.files:
+                saveFileToFilesystem = None
                 if self.eventId:
                     attachmentName = str(saveOriginalFile.id) + "/" +  file.filename
 
                     # isFileInEvent checks if the attachment exists in the database under that eventId and filename.
-                    isFileInEvent = AttachmentUpload.select().where(AttachmentUpload.event == self.eventId,
+                    isFileInEvent = AttachmentUpload.select().where(AttachmentUpload.event_id == self.eventId,
                                                                     AttachmentUpload.fileName == attachmentName).exists()
                     if not isFileInEvent:
                         AttachmentUpload.create(event = self.eventId, fileName = attachmentName)
 
                         # Only save the file if our event is on its own, or the first of a recurring series
-                        if saveOriginalFile and saveOriginalFile.id == self.eventId:       
-                            file.save(self.getFileFullPath(newfilename = attachmentName))
+                        if saveOriginalFile and saveOriginalFile.id == self.eventId:
+                            saveFileToFilesystem = attachmentName
  
                 elif self.courseId:
                     isFileInCourse = AttachmentUpload.select().where(AttachmentUpload.course == self.courseId, AttachmentUpload.fileName == file.filename).exists()
                     if not isFileInCourse:
                         AttachmentUpload.create(course = self.courseId, fileName = file.filename)
-                        file.save(self.getFileFullPath(newfilename = file.filename))
+                        saveFileToFilesystem = file.filename
+                
+                # Creating directory and saving file for events and courses
+                if saveFileToFilesystem:
+                    self.makeDirectory()
+                    file.save(self.getFileFullPath(newfilename = saveFileToFilesystem))        
+                        
         except AttributeError: # will pass if there is no attachment to save
+            
```

Comment:
remove the newline here. That will connect the pass to the except better, visually